sklearn/compose/_column_transformer.py

```diff
@@ -1344,7 +1344,11 @@ def _is_empty_column_selection(column):
     boolean array).
 
     """
-    if hasattr(column, "dtype") and np.issubdtype(column.dtype, np.bool_):
+    if (
+        hasattr(column, "dtype")
+        and isinstance(column.dtype, np.dtype)
+        and np.issubdtype(column.dtype, np.bool_)
```

Comment:
An alternative could be to check for `column.dtype.kind == "b" (the current code would also not work for the pandas boolean dtype)